core/dbt/parser/unit_tests.py

```diff
@@ -105,44 +106,52 @@ def parse_unit_test_case(self, test_case: UnitTestDefinition):
               - {id: 2, b: 2}
         """
         # Add the model "input" nodes, consisting of all referenced models in the unit test.
-        # This creates a model for every input in every test, so there may be multiple
-        # input models substituting for the same input ref'd model.
+        # This creates an ephemeral model for every input in every test, so there may be multiple
+        # input models substituting for the same input ref'd model. Note that since these are
+        # always "ephemeral" they just wrap the tested_node SQL in additional CTEs. No actual table
+        # or view is created.
         for given in test_case.given:
             # extract the original_input_node from the ref in the "input" key of the given list
             original_input_node = self._get_original_input_node(given.input, tested_node)
 
-            original_input_node_columns = None
-            if (
-                original_input_node.resource_type == NodeType.Model
-                and original_input_node.config.contract.enforced
-            ):
-                original_input_node_columns = {
-                    column.name: column.data_type for column in original_input_node.columns
-                }
-
-            # TODO: include package_name?
             input_name = f"{unit_test_node.name}__{original_input_node.name}"
-            input_unique_id = f"model.{package_name}.{input_name}"
-            input_node = ModelNode(
-                raw_code=self._build_fixture_raw_code(
-                    given.get_rows(
-                        self.root_project.project_root, self.root_project.fixture_paths
-                    ),
-                    original_input_node_columns,
+            project_root = self.root_project.project_root
+            common_fields = {
+                "resource_type": NodeType.Model,
+                "package_name": package_name,
+                "original_file_path": original_input_node.original_file_path,
+                "unique_id": f"model.{package_name}.{input_name}",
```

Comment:
I think we may need to include `source_name` in `input_name` to avoid clobbering sources with the same table_name but different source_names when they are inserted to `manifest.nodes` and `manifest.sources` below.

Reply:
Good point.

Reply:
I've changed this to include the source_name. This does make for pretty long unique_ids.  Do we have any concerns about that?  It's not like we're using that name to construct tables or anything...

Reply:
So far I've noticed this issue creep up in https://github.com/dbt-labs/dbt-core/issues/9015. I think we could shorten the node name for CTE generation (since it doesn't need to be unique) but keep the unique_id longer